src/main/java/com/commercetools/web/bind/annotation/PostRequestJsonMapping.java

```diff
@@ -0,0 +1,41 @@
+package com.commercetools.web.bind.annotation;
+
+import org.springframework.core.annotation.AliasFor;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+/**
+ * Custom annotation for {@code POST} endpoints which response (produce) <i>application/json</i> mime-type.
+ * Request type could be any (<i>consume</i> property)
+ *
+ * @see PostJsonRequestJsonMapping
+ */
+@Target({ElementType.METHOD, ElementType.TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+@RequestMapping(method = RequestMethod.POST, produces = APPLICATION_JSON_VALUE)
```

Comment:
You can also use `@GetMapping`

Reply:
> Error: `@PostMapping` and `@GetMapping` annotation type not applicable to this kind of declaration

They both have `@Target(ElementType.METHOD)`

Reply:
💥